politeiawww/backend.go

```diff
@@ -1542,6 +1556,19 @@ func (b *backend) ProcessProposalDetails(propDetails www.ProposalsDetails, isUse
 func (b *backend) ProcessComment(c www.NewComment, user *database.User) (*www.NewCommentReply, error) {
 	log.Debugf("ProcessComment: %v %v", c.Token, user.ID)
 
+	// Verify user used correct key
+	id, ok := database.ActiveIdentity(user.Identities)
```

Comment:
This logic could all go within the `checkSig()` function, couldn't it?

Reply:
Let's me think about this.

Reply:
I suggest putting these 2 small blocks of code into a function called checkPK() instead. Changing checkSig() is not natural to me since checkSig() was originally designed to check the validity of signature instead of public key. Also, to put the logic within checkSig() requires changing parameter arguments.

Reply:
Yes